dbcfeederlib/dbc2vssmapper.py

```diff
@@ -426,6 +426,10 @@ def _analyze_signal(self, expanded_name, node):
         if dbc2vss_def is not None:
             self._analyze_dbc2vss(expanded_name, node, dbc2vss_def)
         if "vss2dbc" in node:
+            if node["type"] != "actuator":
+                # vss2dbc is handled by subscription to target value, so only makes sense
+                # for actuators
+                log.warning("VSS signal %s is not an actuator, vss2dbc not relevant to use", expanded_name)
```

Comment:
I am not sure what this code does or want to say.

"relevant to use"

The menaing is "vss2dbc only supported for actuators", right? However if it is right, this code does nothing, it just prints something, shouldn"t it ignore such amppings then?

So I not fully get what happens actually if there is a mapping of vss2dbc for a sensor

Reply:
Below is the code that manages subscription. As of today I do not think that you get an error if trying to subscribe to target value for a sensor/attribute. (_Alternatively we discard any error message we receive, but I do not think that is the case._)

This code gives a warning if using `vss2dbc` on a sensor or attribute, but it does not ignore the `vss2dbc` statement or give an error. So it is just a hint to the user that they maybe should refactor their mapping file. We could be more intrusive:

- We could ignore the `vss2dbc` after giving warning. Practical behavior would be the same as we anyway do not get any updates for sensors/attributes. The only difference would be if there are errors inside the `vss2dbc` statement, today that would result in errors, if we ignore it they would be no errors either.
- We could give an error and exit.

And we could rephrase the warning but keep the rest intact if we would like. Any opinion?

```
async def subscribe(self, vss_names: List[str], callback):
        """Create a subscription and invoke the callback when data received."""
        entries: List[SubscribeEntry] = []
        for name in vss_names:
            # Always subscribe to target
            subscribe_entry = SubscribeEntry(name, View.FIELDS, [Field.ACTUATOR_TARGET])
            log.info("Subscribe entry: %s", subscribe_entry)
            entries.append(subscribe_entry)

        # If there is a path VSSClient will request a secure connection
        if self._tls and self._root_ca_path:
            root_path: Optional[Path] = Path(self._root_ca_path)
        else:
            root_path = None

        async with VSSClient(self._ip, self._port, token=self._token,
                             root_certificates=root_path, tls_server_name=self._tls_server_name) as client:
            async for updates in client.subscribe(entries=entries):
                log.debug(f"Received update of length {len(updates)}")
                await callback(updates)

```